docs/shared/connectors/metaMask.md

```diff
@@ -10,11 +10,30 @@ Connector for the [MetaMask SDK](https://github.com/MetaMask/metamask-sdk).
 
 ## Import
 
-::: warning WARNING This connector has a large file size due to the underlying [`@metamask/sdk`](https://github.com/MetaMask/metamask-sdk). 
-For mobile support, it is recommended to use <a :href="`/${docsPath}/api/connectors/walletConnect`">`walletConnect`</a>. For desktop support, you should rely on <a :href="`/${docsPath}/api/createConfig#multiinjectedproviderdiscovery`">Multi Injected Provider Discovery</a> (EIP-6963) via the Wagmi <a :href="`/${docsPath}/api/createConfig#config`">`Config`</a>.
-
-On desktop, if you must target MetaMask, you can use <a :href="`/${docsPath}/api/connectors/injected`">`injected`</a>, e.g. <a :href="`/${docsPath}/guides/migrate-from-v1-to-v2#removed-metamaskconnector`">`injected({ target: 'metaMask' })`</a>. Please keep in mind this approach does not use EIP-6963 and other injected wallets could conflict with the connector.
-:::
+>[!WARNING]  
+> This connector has a large file size due to the underlying @metamask/sdk.
+
+>[!WARNING]  
+> MetaMask SDK can be used for mobile and desktop support.
+
+>[!WARNING]  
+> MetaMask SDK supports EIP-6963, which prevents other injected wallets to conflict with the connector.
+
+>[!WARNING]
+> Due to a limitation in iOS, Safari cancels deep link connections if they do not happen within 500ms. It's recommended to use a direct RPC method for transactions to avoid this issue. Instead of using `getSigner` and `sendTransaction`, directly invoke the `send` method with `'eth_sendTransaction'` and wait for the transaction using `waitForTransaction`. This approach avoids the deep link timeout issue on iOS Safari.
+>
+> **Implementation with potential timeout issue:**
+> ```js
+> const signer = provider.getSigner()
+> const txSubmitRes = await signer.sendTransaction(transactionData)
+> ```
+>
+> **Recommended Implementation (to avoid timeout):**
```

Comment:
I’d suggest a recommended implementation would be to estimate the gas first via `useEstimateGas`, then pass that to `useSendTransaction` to avoid another async request (if we can’t favor deep links over universal links).

Reply:
or set `gas: null` in `useSendTransaction` to skip gas estimation entirely.